examples/Tabs.overflow.tsx

```diff
@@ -3,50 +3,93 @@
-      case 3:
-        return 'The company re-invests 20% of their revenues in research and development.';
-      case 4:
-        return 'Bentley Systems is headquartered in Exton, Pennsylvania, United States, but has development, sales and other departments in over 50 countries.';
-      case 5:
-        return 'The company had revenues of $700 million in 2018.';
-      case 6:
-        return 'Keith A. Bentley and Barry J. Bentley founded Bentley Systems in 1984.';
-      default:
-        return 'They introduced the commercial version of PseudoStation in 1985, which allowed users of Intergraphs VAX systems to use low-cost graphics terminals to view and modify the designs on their Intergraph IGDS (Interactive Graphics Design System) installations.';
-    }
-  };
-  const labels = [
-    <Tab key={1} label='Bentley Overview' />,
-    <Tab key={2} label='Markets' />,
-    <Tab key={3} label='Uses' />,
-    <Tab key={4} label='Reinvestment' />,
-    <Tab key={5} label='Location' />,
-    <Tab key={6} label='Revenue' />,
-    <Tab key={7} label='Founders' />,
-    <Tab key={8} label='History' />,
-  ];
+  const [activeIndex, setActiveIndex] = React.useState(0);
+
+  const [tabData, setTabData] = React.useState([
+    { name: 'Bentley Overview', active: true },
+    { name: 'Markets', active: false },
+    { name: 'Uses', active: false },
+    { name: 'Reinvestment', active: false },
+    { name: 'Location', active: false },
+    { name: 'Revenue', active: false },
+    { name: 'Founders', active: false },
+    { name: 'History', active: false },
+  ]);
 
   return (
     <div style={{ maxWidth: 425, border: '1px solid lightpink', padding: 8 }}>
-      <Tabs
-        labels={labels}
-        overflowOptions={{ useOverflow: true }}
-        onTabSelected={setIndex}
-        activeIndex={index}
-      >
-        {getContent()}
+      <Tabs overflowOptions={{ useOverflow: true }}>
+        <Tabs.TabList>
+          {tabData?.map((item, index) => {
+            return (
+              <Tabs.Tab
+                isActive={item.active}
+                label={item.name}
+                onActiveChange={() => {
+                  setTabData((oldData) => {
+                    const newData = [...oldData];
+
+                    const oldActiveTab = { ...newData[activeIndex] };
+                    oldActiveTab.active = false;
+                    newData[activeIndex] = oldActiveTab;
+
+                    const newActiveTab = { ...newData[index] };
+                    newActiveTab.active = true;
+                    newData[index] = newActiveTab;
+
+                    return newData;
+                  });
+                  setActiveIndex(index);
+                }}
+              />
+            );
+          })}
+        </Tabs.TabList>
+
+        <Tabs.Panels>
+          <Tabs.Panel>
```

Comment:
How users know which TabPanel goes with which Tab?

Reply:
I figured users would list their `Panel` elements in the same order as their `Tab` elements in order to associate the two.

Reply:
on the outside, the association might be based on order, but on the inside, it needs to be explicit through the use of `id`/`aria-controls`. see APG: https://www.w3.org/WAI/ARIA/apg/patterns/tabs/ (you can inspect the examples)

it might make sense to ask for a unique value from the users, to explicitly connect the tabs and panels from the outside. not sure about the api 🤔

Reply:
The `value` prop has been added to `Tab` and `Panel` subcomponents. As a result, an `index` value is no longer passed through context meaning `TabsPanelsContext` has been removed and `TabListContext` does not wrap each `Tab` element!

Reply:
Maybe value is not the most intuitive name for this functionality.